src/renderers/dom/client/ReactEventListener.js

```diff
@@ -60,29 +62,59 @@ PooledClass.addPoolingTo(
 );
 
 function handleTopLevelImpl(bookKeeping) {
-  var topLevelTarget = ReactMount.getFirstReactDOM(
-    getEventTarget(bookKeeping.nativeEvent)
-  ) || window;
-
-  // Loop through the hierarchy, in case there's any nested components.
-  // It's important that we build the array of ancestors before calling any
-  // event handlers, because event handlers can modify the DOM, leading to
-  // inconsistencies with ReactMount's node cache. See #1105.
-  var ancestor = topLevelTarget;
-  while (ancestor) {
-    bookKeeping.ancestors.push(ancestor);
-    ancestor = findParent(ancestor);
-  }
+  var path = bookKeeping.nativeEvent.path;
+  var topLevelTargetID;
+  if (path != null) {
+    var oldRootId = undefined;
+    var currentNativeTarget = undefined;
+    for (var j = 0; j < path.length; j++) {
```

Comment:
Can we name `path[j]` to something in the local variable? Some of the lines below doesn't read very nicely because you think "wtf is path[j] again?"